terraform/modules/k8s-karpenter/main.tf

```diff
@@ -144,8 +148,6 @@ resource "helm_release" "this" {
   version             = local.karpenter.chart_version
   namespace           = module.namespace[count.index].name
   max_history         = 3
-  repository_username = data.aws_ecrpublic_authorization_token.token.user_name
```

Comment:
I'm not sure it works. Let's test it together

Reply:
I tested this yesterday in the sandbox account, and in the  Propeller account. All works fine. More over, this data works only for `us-east-1` region. For other regions we need to use additional aws provider with alias.